### PR TITLE
Removing congrats screen from plan purchases

### PR DESCRIPTION
**Pull request by lsl**
Fixes;
- https://github.com/Automattic/dotcom-forge/issues/9183

Related to;
- https://github.com/Automattic/wp-calypso/pull/94766 
- https://github.com/Automattic/wp-calypso/pull/94929

## Proposed Changes

* Following a similar pattern as in https://github.com/Automattic/wp-calypso/pull/94766
* When only plan purchases are present in the cart, take the user directly to /plans.

<details><summary> 📝 TODO from @lsl </summary>
<p>

- Add tests for thank you function changes
- Confirm whether this should apply to onboarding flows
- Confirm if this should apply to domain + plan purchases, I believe so.
- I'm proposing a thank you notice component in this PR, need to decide if we should keep it or not.

</p>
</details>

## Testing Instructions

Plan upgrades:
* Purchase a plan upgrade from /plans
* You should no longer see the thank you page; instead, go directly to the plans screen.
* You should see a toast notification temporarily.

No changes:
1. Purchases including non-plan products should continue to show a thank you page
1. Plan purchases during sign-up/onboarding should continue to show ~~a thank you page~~ the Goals step
1. eCommerce plan purchase, as there is a special logic for that: https://github.com/Automattic/wp-calypso/blob/f4188f019a5de25a2f0f443a12b7602a69ba7d96/client/my-sites/checkout/get-thank-you-page-url/index.ts#L452-L473
1. Jetpack plan purchases: https://github.com/Automattic/wp-calypso/blob/f4188f019a5de25a2f0f443a12b7602a69ba7d96/client/my-sites/checkout/get-thank-you-page-url/index.ts#L545-L572

### CODE REVIEW

File: client/my-sites/checkout/get-thank-you-page-url/index.ts [hunk 1]
```diff
@@ -615,6 +616,12 @@ function getFallbackDestination( {
 		);
 	}
 
+	const planItems = cart?.products?.filter( ( product ) => isWpComPlan( product.product_slug ) );
+	if ( planItems && planItems.length > 0 && planItems.length === cart?.products?.length ) {
+		debug( 'site with plan product' );
+		return `/plans/${ siteSlug }?success=${ planItems[ 0 ].product_slug }`;
```
Comment (lsl) [inline client/my-sites/checkout/get-thank-you-page-url/index.ts]: Should be /plans/my-plan/:site

File: client/my-sites/plans/current-plan/controller.jsx [hunk 2]
```diff
@@ -20,7 +20,9 @@ export function currentPlan( context, next ) {
 	const isEcommerceTrial = currentPlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
 	const isEntrepreneurTrial = isEcommerceTrial && ! purchase?.isWooExpressTrial;
 
-	if ( isFreePlan || isEntrepreneurTrial ) {
+	// When redirecting from the checkout page, selectedSite's plan hasn't had time to update yet, use success param to
+	// to bypass the isFreePlan check here and allow loading of /plans/my-plan.
+	if ( ( ! context.query.hasOwnProperty( 'success' ) && isFreePlan ) || isEntrepreneurTrial ) {
 		page.redirect( `/plans/${ selectedSite.slug }` );
 
 		return null;
```
Comment (lsl) [inline client/my-sites/plans/current-plan/controller.jsx:27]: Might remove this and go back to /plans, /plans/my-plan feels out of date and the same info is on the plans page now too.

**Review by fushar — DISMISSED** [review]
✅ Plan upgrade:

image

---

✅ Purchases including non-plan products should continue to show a thank you page

image

---

✅ Plan purchases during sign-up/onboarding should ~~continue to show a thank you page~~

- I believe you meant it should continue to the Goals step as expected.
---

✅ eCommerce plan purchase, as there is a special logic for that

image

image

When the CTA button is clicked, we're taken here. Maybe in the future, we could just redirect directly to here instead? 🤔 

image

---

✅ Jetpack plan purchases

image

**Review by fushar — COMMENTED** [review]
Upgrading from Free to Premium seems to still show the congrats page 🤔 

image

**Comment by okmttdhr** [comment]
Thank you for your tests!! 🥺

>I believe you meant it should continue to the Goals step as expected.

Sorry, I updated the description!

>When the CTA button is clicked, we're taken here. Maybe in the future, we could just redirect directly to here instead? 🤔

Oh, that would be a good improvement! I'll try to address this if possible. CC: @lsl

>Upgrading from Free to Premium seems to still show the congrats page 🤔

To confirm, did you go through the plan upgrade offer page before the congrats page, @fushar?

**Comment by fushar** [comment]
> To confirm, did you go through the plan upgrade offer page and click "No Thanks" before the congrats page, @fushar?

Correct, I saw that offer and clicked "No Thanks".

**Review by fushar — APPROVED** [review]

File: client/my-sites/plans/current-plan/controller.jsx (hunk 2 above)
Comment (lsl) [reply client/my-sites/plans/current-plan/controller.jsx:27]: I think we can remove this now.

**Comment by lsl** [comment]
> Upgrading from Free to Premium seems to still show the congrats page

Similar issue for me going from free to personal.

**Comment by okmttdhr** [comment]
Closing this PR since we need to revisit the whole checkout flow: pbxlJb-6kE-p2#comment-3936

>Upgrading from Free to Premium seems to still show the congrats page

For this particular debugging, see p1728629173664729/1728622995.246169-slack-CRWCHQGUB

**Pull request closed without merging**